lib/ansible/modules/cloud/amazon/aws_secret_facts.py

```diff
@@ -0,0 +1,203 @@
+    returned: always
+    type: boolean
+rotation_lambda_arn:
+    description:
+      - The ARN of an AWS Lambda function that's invoked by Secrets Manager to rotate and expire the secret either automatically per
+        the schedule or manually by a call to RotateSecret .
+    returned: always
+    type: string
+rotation_rules:
+    description: A structure that defines the rotation configuration for the secret.
+    returned: always
+    type: dict
+automatically_after_days:
+    description: Specifies the number of days between automatic scheduled rotations of the secret.
+    returned: always
+    type: int
+last_rotate_date:
+    description: The last date and time that the rotation process for this secret was invoked.
+    returned: always
+    type: string
+last_changed_date:
+    description: The last date and time that this secret was modified in any way.
+    returned: always
+    type: string
+last_access_date:
+    description: The last date that this secret was accessed. This value is truncated to midnight of the date and therefore shows only the date, not the time.
+    returned: always
+    type: string
+deleted_date:
+    description:
+      - The date and time on which this secret was deleted. Not present on active secrets. The secret can be recovered until the number of days
+        in the recovery window has passed, as specified in the RecoveryWindowInDays parameter of the DeleteSecret operation.
+    returned: always
+    type: string
+tags:
+    description: The list of user-defined tags that are associated with the secret.
+    returned: always
+    type: list
+secret_versions_to_stages:
+    description: A list of all of the currently assigned SecretVersionStage staging labels and the SecretVersionId that each is attached to.
+    returned: always
+    type: dict
+'''
+
+import os
+import re
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def camel_list_to_snake_list(list):
+    conv_list = []
+    for i in list:
+        conv_list.append(camel_dict_to_snake_dict(i))
+    return conv_list
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'name': dict(type='str'),
+            'prefix': dict(type='str'),
+            'suffix': dict(type='str'),
+            'regex': dict(type='str'),
+        },
+        supports_check_mode=True,
+    )
+
+    result = {
+        'changed': True,
+        'desired_secrets': []
+    }
+
+    client = module.client('secretsmanager')
+
+    kwargs = {}
+    all_secrets = []
+
+    try:
+        while True:
+            response = client.list_secrets(**kwargs)
+            for i in response['SecretList']:
+                all_secrets.append(i)
+            try:
+                kwargs['nextToken'] = response['nextToken']
```

Comment:
I think this should also break if `nextToken` is None/False/empty to make sure this loop doesn't run a final query and duplicate secrets.

Reply:
I think 'nextToken' is only in the response if there are more results than what was returned by the last call (according to the docs), so breaking on the KeyError should be fine.